### PR TITLE
Fix generic types on Return type for s3 waiters

### PR DESCRIPTION
**Pull request by jan-xyz**

### CODE REVIEW

File: aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/waiters/AmazonS3Waiters.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public AmazonS3Waiters(AmazonS3 client) {
      * entered the desired state or not, where polling criteria is bound by
      * either default polling strategy or custom polling strategy.
      */
-    public Waiter bucketNotExists() {
+    public Waiter<HeadBucketRequest> bucketNotExists() {
```
Comment (dagnir) [inline aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/waiters/AmazonS3Waiters.java]: Can we all add the generic parameters for the acceptors? (Like `HttpFailureStatusAcceptor<HeadBucketResult>`?)

**Comment by jan-xyz** [comment]
I pushed the changes for the `Http*StatusAcceptors`, there remains another warning, I'm not sure how to fix that.

**Comment by jan-xyz** [comment]
I get `unchecked generic array creation for varargs parameter` on the `withAcceptor` call. I was thinking to solve it like it is solved in other SDKs (e.g. the [aws-java-sdk-emr](https://github.com/aws/aws-sdk-java/tree/4734de6fb0f80fe5768a6587aad3b9d0eaec388f/aws-java-sdk-emr/src/main/java/com/amazonaws/services/elasticmapreduce/waiters)), with one class per waiter state, but the code in there is generated. I could build it like that but I'm not sure if it's the right approach and on my first google I couldn't find how the code generator works.

**Comment by dagnir** [comment]
Hmm, I think this actually safe so that should fine; we don't depend on the the type of the temporary array.

**Comment by jan-xyz** [comment]
Okay, is there anything else I can do?

**Comment by shorea** [comment]
Merged, thanks for the contribution!

**Pull request merged**